lib/virtual-fs/src/host_fs.rs

```diff
@@ -16,10 +16,22 @@ use std::task::{Context, Poll};
 use std::time::{SystemTime, UNIX_EPOCH};
 use tokio::fs as tfs;
 use tokio::io::{AsyncRead, AsyncSeek, AsyncWrite, ReadBuf};
+use tokio::runtime::Handle;
 
-#[derive(Debug, Default, Clone)]
```

Comment:
Putting a tokio handle on the FS, and then again on each file, and the trying to get the current env or using a handle seems... like an extremely leaky abstraction and hacky.

Can't we make the callers (eg syscalls) ensure that a runtime is available, and just make these methods async?

Reply:
The tokio handle is extremely brittle, this is a better way of doing it - keeps the tokio abstraction largely away from the WASM threads. Syscalls should not be aware of tokio at all... in the browser we won't be using the tokio handles anyway so it should not be in the syscalls